raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -633,6 +634,47 @@ public void testBeginQuorumEpochHeartbeat(boolean withKip853Rpc) throws Exceptio
         context.assertSentBeginQuorumEpochRequest(epoch, Set.of(remoteId1, remoteId2));
     }
 
+    @Test
+    public void testBeginQuorumShouldNotSendAfterFetchRequest() throws Exception {
```

Comment:
let's also add/modify a code comment in `testBeginQuorumEpochHeartbeat` 

```
// begin epoch requests should be sent out every beginQuorumEpochTimeoutMs
```
->
```
// begin epoch requests sent out every beginQuorumEpochTimeoutMs if replicas have not fetched
```